master/trials.tx.up.sql

```diff
@@ -0,0 +1,76 @@
+-- keeping out of the migration directory for now to avoid having it run
+ALTER table trials ADD COLUMN IF NOT EXISTS tags jsonb DEFAULT '{}';
+-- if similar data structure were to be used where the actual values of the tags
+-- was considered, index should not use jsonb_path_ops (just remove that part)
+CREATE INDEX trials_tags_index ON trials USING GIN (tags jsonb_path_ops);
+-- update trials set tags = '{}';
+
+CREATE TABLE trials_collections (
+    -- table to store a set of filters as defined in QueryFilters in api/trial.proto
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,	
+    user_id integer REFERENCES public.users(id) ON DELETE CASCADE NOT NULL,
+    project_id integer REFERENCES public.projects(id) ON DELETE CASCADE NOT NULL,
+	  name text NOT NULL,
+	  filters jsonb NOT NULL,
+    sorter  jsonb NOT NULL,
+  CONSTRAINT project_filter_name_uniq UNIQUE (project_id, name)
+);
+
+-- make analogous modifications to those found in reindex-steps pr
+-- because we are joining validations on (trial_id, total_batches)
+CREATE INDEX validations_archived ON raw_validations(archived);
+ALTER TABLE raw_validations DROP CONSTRAINT validations_trial_id_run_id_total_batches_unique;
+CREATE UNIQUE INDEX validations_trial_id_total_batches_run_id_unique ON raw_validations(trial_id, total_batches, trial_run_id);
+
+
+CREATE AGGREGATE jsonb_collect(jsonb) (
+  SFUNC = 'jsonb_concat',
+  STYPE = jsonb,
+  INITCOND = '{}'
+);
+
+
+-- aggregations should be mostly redundant.the only way they 
+-- would get applied is when there are multiple trial_run_ids for a single 
+-- (trial_id, total_batches) on steps or validations. in which case,
+-- which trial_run_id is it appropriate to take the metrics from?
+-- there have been situations where they differ.
```

Comment:
basically in order to get rid of the aggregation,  I am blocked on the question "Which `trial_run_id` is the proper source of truth for metrics."

I have seen situations where for the same `(trial_id, total_batches)`, run 2 has `{"metric_a", "0.33", "metric_b": 0.66}` and run 3 has {`"metric_a": "0.33"}` (which means I cant just take the first or the last).